package-builder/flox-build.mk

```diff
@@ -218,7 +218,8 @@ endef
 # to the inner "build wrapper" environment in the in-situ build mode in support
 # of the tools and compilers found in the outer environment, and for Flox to
 # otherwise function properly.
-ALLOW_OUTER_ENV_VARS := FLOX_RUNTIME_DIR HOME PATH \
+ALLOW_OUTER_ENV_VARS := FLOX_ACTIVATE_TRACE FLOX_ENV_DIRS FLOX_RUNTIME_DIR HOME PATH CPATH \
+  PKG_CONFIG_PATH LIBRARY_PATH DYLD_FALLBACK_LIBRARY_PATH \
```

Comment:
Does this need to match anything set by https://github.com/flox/flox/blob/main/assets/environment-interpreter/common/etc/profile.d/0100_common-paths.sh? When else might we need to remember to update it?

Reply:
_Probably yes ..._ I iteratively came up with this list by working through the manifest build examples repo, but planned to circle back to look for logical explanations for these, and then to hopefully infer the other variables we should be similarly passing from outer --> inner activations.

Reply:
I went through the list of variables exported from all etc-profile scripts and updated the `ALLOW_OUTER_ENV_VARS` variable to allow them all through. Then I started testing and found why you can't do that:

- letting `LIBRARY_PATH` through causes the resulting executables to have references to both the build and develop environments in their `RUNPATH`

Still looking for more exceptions ... so far the result can be seen in commit f2db8df69380b2b61cce5459e5b15b36d7e52507 (still being pushed by infinitely slow ssh transport from flight ... I'm guessing the provider is doing stateful inspection and nobbling of ssh traffic to prevent it from being used to tunnel netflix, but damn they've _really_ nobbled it!).

Reply:
Builds are working now with the expanded list of variables to be passed so I'll go ahead and resolve the conversation.

The real test will come from creating a large battery of test builds to exercise all the known things, at which point we can reassess the completeness of the list and/or the question of whether it's better to adopt an subtractive (deny list) approach as opposed this additive (allow list) approach.